### DISCUSSION TITLE
File sharing between two systems which are in two private networks

### DISCUSSION THREAD

**Discussion by sarath3192** [n0-computer/iroh#693] [General]
We are trying to transfer file between two systems which are in two private network using iroh-share example. Is this possible without port forwarding?

**Comment by flub** [comment]
Hi, iroh-share so far has only been targetted at a local network.  The idea was to add remote connections with holepunching etc.  But this won't happen anytime soon, we're concentrating on other parts at the moment.  So port-forwarding is probably what works for now.

**Reply by sarath3192** [reply]
Thank you for detailed reply @flub

**Comment by imrank03** [comment]
> Hi, iroh-share so far has only been targetted at a local network. The idea was to add remote connections with holepunching etc. But this won't happen anytime soon, we're concentrating on other parts at the moment. So port-forwarding is probably what works for now.

Hi @flub, but on [IPFS-desktop](https://github.com/ipfs/ipfs-desktop)  I haven't port-forwarded, still it works on two different networks. Are they using holepunching?

**Reply by flub** [reply]
Probably, I'm not familiar with ipfs-desktop which is another project.  IPFS has holepunching in the spec and kubo does some forms of holepunching.

**Reply by jxs** [reply]
`ipfs-desktop` uses [`kubo`](https://github.com/ipfs/kubo) which uses `go-libp2p` [which supports UPNP](https://github.com/libp2p/go-libp2p/blob/313b080ea4e27f47dbfb9f872133b9fba4a9d183/options.go#L418) whereas `rust-libp2p` [doesn't yet](https://github.com/libp2p/rust-libp2p/issues/558).

**Comment by b5** [comment]
Iroh has had the ability to auto-holepunch across networks for some time now, I'm going to close this as stale
